opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -490,11 +490,59 @@ static int mca_btl_tcp_component_close(void)
  *  Create a btl instance and add to modules list.
  */
 
-static int mca_btl_tcp_create(int if_kindex, const char* if_name)
+static int mca_btl_tcp_create(const int if_kindex, const char* if_name)
 {
     struct mca_btl_tcp_module_t* btl;
     char param[256];
     int i;
+    struct sockaddr_storage addr;
+    bool found = false;
+
+    /*
```

Comment:
I am not sure we are not overdoing it. We have the unique name of the iface we want to use, why are we trying to play around parsing all interfaces in order to find the correct kindex instead of just using the if_name via opal_ifnametokindex ?